sklearn/utils/estimator_checks.py

```diff
@@ -553,6 +554,45 @@ def check_sample_weights_list(name, estimator_orig):
         estimator.fit(X, y, sample_weight=sample_weight)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_sample_weight_invariance(name, estimator_orig):
+    if (has_fit_parameter(estimator_orig, "sample_weight") and
+                    name not in ["KMeans", "MiniBatchKMeans"]):
+        estimator1 = clone(estimator_orig)
+        estimator2 = clone(estimator_orig)
+
+        X = np.array([[1, 3], [1, 3], [1, 3], [1, 3],
+                      [2, 1], [2, 1], [2, 1], [2, 1],
+                      [3, 3], [3, 3], [3, 3], [3, 3],
+                      [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.dtype('float'))
+        y = np.array([1, 1, 1, 1, 2, 2, 2, 2,
+                      1, 1, 1, 1, 2, 2, 2, 2], dtype=np.dtype('float'))
+
+        if has_fit_parameter(estimator_orig, "random_state"):
+            estimator1.fit(X, y=y, sample_weight=np.ones(shape=len(y)), random_state=0)
+            estimator2.fit(X, y=y, sample_weight=None, random_state=0)
+        else:
+            estimator1.fit(X, y=y, sample_weight=np.ones(shape=len(y)))
+            estimator2.fit(X, y=y, sample_weight=None)
+
+        if hasattr(estimator_orig, "predict"):
+            X_pred1 = estimator1.predict(X)
+            X_pred2 = estimator2.predict(X)
+            try:
+                assert_allclose(X_pred1, X_pred2, rtol=0.5)
```

Comment:
maybe you might want to use `assert_allclose_dense_sparse` if the output could be sparse.

Reply:
pass the err_msg to assert_allclose_dense_sparse avoiding the try .. except ... to raise the proper error.

Reply:
I don't think we expect the output to be sparse. 
err_msg is done!